### PR TITLE
Fixed invoice loading duration issue by removing unnecessary es calls

### PR DESCRIPTION
**Pull request by prasanthchaduvula**
## Notion card
https://www.notion.so/Fix-slow-loading-times-on-Invoices-page-382c353f4fd2454d9fa0985bd23cc2a7
## Summary

- Used memorization and removed unnecessary multiple ES calls being made in one API.
- Moved recently updated invoices, from PG to ES.

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactor (non-breaking and retains same functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to
      not work as expected)
- [ ] This change requires a documentation update

### Checklist:

- [ ] I have manually tested all workflows
- [ ] I have performed a self-review of my own code
- [ ] I have annotated changes in the PR that are relevant to the reviewer
- [ ] I have added automated tests for my code

### CODE REVIEW

**Comment by supriya3105** [comment]
@shalapatil @apoorv1316 Please review and test this on priority. The invoice page load time should improve with the fix in this PR. 

If yes, let's deploy to staging and release tomorrow.

File: app/services/invoices/index_service.rb [hunk 1]
```diff
@@ -37,10 +37,15 @@ def invoices_query
       end
 
       def recently_updated_invoices
-        current_company.invoices.kept
```
Comment (shalapatil) [inline app/services/invoices/index_service.rb:40]: do we need to consider kept while querying  from ES?

**Review by shalapatil — COMMENTED** [review]
LG except one comment

File: app/services/invoices/index_service.rb (hunk 1 above)
Comment (prasanthchaduvula) [reply app/services/invoices/index_service.rb:40]: @shalapatil Here we want to query on kept invoices only. 
```
Invoice.search(
          where: filters.where_clause,
          order: { updated_at: :desc },
          limit: 10,
          includes: [:client]
        )
```
Here in the where clause we have { company_id: , discarded_at: nil)

**Comment by supriya3105** [comment]
@shalapatil Please approve, otherwise we won't be able to merge the PR.

**Review by apoorv1316 — APPROVED** [review]

**Pull request merged**